src/shogun/mathematics/linalg/LinalgBackendEigen.h

```diff
@@ -284,6 +300,28 @@ class LinalgBackendEigen : public LinalgBackendBase
 		result_eig = alpha * a_eig + beta * b_eig;
 	}
 
+	/** Eigen3 Cholesky decomposition */
+	template <typename T>
+	SGMatrix<T> cholesky_impl(const SGMatrix<T>& a, const bool lower) const
+	{
+		SGMatrix<T> c(a.num_rows, a.num_cols);
+		set_const_impl<T>(c, 0);
+		typename SGMatrix<T>::EigenMatrixXtMap a_eig = a;
+		typename SGMatrix<T>::EigenMatrixXtMap c_eig = c;
+
+		Eigen::LLT<Eigen::Matrix<T,Eigen::Dynamic,Eigen::Dynamic> > llt(a_eig);
+		//compute matrix L or U
+		if(lower==false)
+			c_eig = llt.matrixU();
+		else
+			c_eig = llt.matrixL();
+
+		// checking for success
+		REQUIRE(llt.info()!=Eigen::NumericalIssue, "Matrix is not Hermitian positive definite!\n");
```

Comment:
Can you comment (in source code) on the possible outcomes of the eigen3 info?
And can you properly translate this into a shogun error message?

@vigsterkr what about some proper exception handling where we forward such numerical errors as numerical errors rather than just standard exceptions that make shogun stop?